### PR TITLE
Store env in key vault and add Parameters to long haul tests

### PR DESCRIPTION
**Pull request by brycewang-microsoft**

### CODE REVIEW

File: e2e/Tests/prerequisites/e2eTestsSetup.ps1 [hunk 1]
```diff
@@ -773,7 +773,7 @@ catch
 CleanUp-Certs
 
 # Creating a file to run to load environment variables
-$loadScriptDir = Join-Path $PSScriptRoot "..\..\..\..\.." -Resolve
```
Comment (brycewang-microsoft) [inline e2e/Tests/prerequisites/e2eTestsSetup.ps1:776]: The file path was refactored previously, so correcting this

File: e2e/LongHaul/device/Parameters.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        Mqtt,
+        Amqp,
+    };
+
+    internal class Parameters
+    {
+        [Option(
+            'a',
+            "ApplicationKey",
```
Comment (drwill-ms) [inline e2e/LongHaul/device/Parameters.cs:19]: InstrumentationKey

File: e2e/LongHaul/device/Parameters.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            Required = true,
+            HelpText = "The instrumentation key string for application insights.")]
+        public string ApplicationKey { get; set; }
```
Comment (drwill-ms) [inline e2e/LongHaul/device/Parameters.cs:22]: Needs env var defaults.

File: e2e/LongHaul/device/Parameters.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace Microsoft.Azure.IoT.Thief.Device
+{
+    public enum TransportType
+    {
+        Mqtt,
+        Amqp,
+    };
+
+    internal class Parameters
```
Comment (drwill-ms) [inline e2e/LongHaul/device/Parameters.cs:15]: Needs device connection string.

File: e2e/LongHaul/device/Program.cs [hunk 5]
```diff
@@ -16,23 +14,30 @@ namespace ThiefDevice
     internal class Program
     {
         private static readonly IDictionary<string, string> s_commonProperties = new Dictionary<string, string>();
-        private static Settings s_settings;
         private static Logger s_logger;
         private static ApplicationInsightsLoggingProvider s_aiLoggingProvider;
+        private static readonly string s_deviceConnectionString = TestConfiguration.DeviceConnectionString;
```
Comment (drwill-ms) [inline e2e/LongHaul/device/Program.cs:19]: Nope, should be created by the ARM template, saved as a key vault secret, and fetched as an env var.

File: e2e/LongHaul/device/TestConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Azure.IoT.Thief.Device
+{
+    public static class TestConfiguration
```
Comment (drwill-ms) [inline e2e/LongHaul/device/TestConfiguration.cs:8]: Remove

File: e2e/LongHaul/prerequisites/LoadEnvironmentVariablesFromKeyVault.ps1 [hunk 7]
```diff
@@ -0,0 +1,53 @@
+param(
```
Comment (drwill-ms) [inline e2e/LongHaul/prerequisites/LoadEnvironmentVariablesFromKeyVault.ps1:1]: This is the same as the file under e2e\Tests\prerequisites, right?

Let's just use the same one. We can move it up to be under e2e directly.

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 [hunk 8]
```diff
@@ -177,7 +184,9 @@ az deployment group create `
     --only-show-errors `
     --template-file "$PSScriptRoot\longhaul-resources.json" `
     --parameters `
+    UserObjectId=$userObjectId `
```
Comment (drwill-ms) [inline e2e/LongHaul/prerequisites/LongHaulSetup.ps1:187]: Why do we need this?

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 [hunk 9]
```diff
@@ -219,8 +229,42 @@ if (-not $longhaulDevice)
 $longhaulDeviceConnectionString = az iot hub device-identity connection-string show --device-id $longhaulDeviceId --hub-name $iotHubName --resource-group $ResourceGroup --output tsv
 
 ############################################################################################################################
-# Print variables
+# Store all secrets in a KeyVault - Values will be pulled down from here to configure environment variables.
+############################################################################################################################
+
+$keyvaultKvps = @{
```
Comment (drwill-ms) [inline e2e/LongHaul/prerequisites/LongHaulSetup.ps1:235]: Should include the AI key too.

File: e2e/Tests/prerequisites/e2eTestsSetup.ps1 (hunk 1 above)
Comment (drwill-ms) [reply e2e/Tests/prerequisites/e2eTestsSetup.ps1:776]: It needs to go up to the directory above the repo root.

So if this file is in e2e/Tests/prerequisites, this should go up 3 dirs, so "..\..\..".

File: e2e/LongHaul/prerequisites/LongHaulSetup.ps1 (hunk 8 above)
Comment (brycewang-microsoft) [reply e2e/LongHaul/prerequisites/LongHaulSetup.ps1:187]: it's been used by [keyvault](https://github.com/Azure/azure-iot-sdk-csharp/blob/brycewang/settings/e2e/LongHaul/prerequisites/longhaul-resources.json#L126).

Comment (drwill-ms) [reply e2e/LongHaul/prerequisites/LongHaulSetup.ps1:187]: Oh, to give the current user permissions to the key vault secrets. Got it.

File: e2e/LongHaul/device/Parameters.cs (hunk 3 above)
Comment (brycewang-microsoft) [reply e2e/LongHaul/device/Parameters.cs]: I couldn't assign env var as defaults here. It's complaining with [CS0182](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0182?f1url=%3FappId%3Droslyn%26k%3Dk(CS0182))

**Pull request merged**